test/controllers/courses_controller_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class CoursesControllerTest < ActionDispatch::IntegrationTest
+
+    def setup
+        @user = User.create(name: "Courses user", email: "user@courses.ca", password: "coursebar", password_confirmation: "coursebar", role: Role.admin)
+        log_in_user(@user, "coursebar")
```

Comment:
Shouldn't enforce a particular logged in user for each test. Better to let each test case handle their own access level. You could move this to a test helper to make it cleaner eg `log_in_as_admin_user()`